### ISSUE TITLE
Changeling DNA absorption bug

### ISSUE THREAD

**Issue by Iamgoofball** [NTStation/NTstation13#363]
_Issue by **[Kuromoi](https://github.com/Kuromoi)** from Saturday Mar 08, 2014 at 19:54 GMT_
_Originally opened as https://github.com/tgstation/-tg-station/issues/3026_

---

Sometimes (and I'm really not sure why), changelings cannot absorb more than five DNA strands even when they transform after absorbing the first set.  This makes greentext literally impossible.

This isn't a new issue either; it's been in place since the new changeling code was added.  I do not know how to reproduce this.

**Comment by Iamgoofball** [comment]
_Comment by **[Razharas](https://github.com/Razharas)** from Saturday Mar 08, 2014 at 21:03 GMT_

---

It happens due to DNA test being just list position fuckery

**Comment by Iamgoofball** [comment]
_Comment by **[Kuromoi](https://github.com/Kuromoi)** from Saturday Mar 08, 2014 at 21:15 GMT_

---

Can it be fixed?  IT's happened to me both times I've played newling (I took a haitus shortly after it was introduced for unrelated reasons)...

**Comment by Iamgoofball** [comment]
_Comment by **[Razharas](https://github.com/Razharas)** from Saturday Mar 08, 2014 at 22:27 GMT_

---

It can be fixed, ill look at it later

**Comment by Iamgoofball** [comment]
_Comment by **[Kuromoi](https://github.com/Kuromoi)** from Monday Mar 10, 2014 at 16:00 GMT_

---

Any progress?

**Comment by Iamgoofball** [comment]
_Comment by **[Razharas](https://github.com/Razharas)** from Monday Mar 10, 2014 at 22:19 GMT_

---

Nope, havent even looked at it yet

**Comment by Iamgoofball** [comment]
_Comment by **[Razharas](https://github.com/Razharas)** from Monday Mar 10, 2014 at 22:48 GMT_

---

Well looked at it and i cant say for sure what is wrong, cant reproduce also

**Comment by validsalad** [comment]
well not sure if this issue is related but

I presumably had no limit for absorbing genomes and was able to suck out eight without needing to transform. Then I used a transformation sting on a monkey and after that, I lost all the genomes that weren't used for that sting as options for the sting.

but wait there's more

there's no check for checking how many times the player has clicked for absorbing dna so

so

I ended up with 255 genomes once.
